zipkin-lens/src/components/DiscoverPage/SearchBar/CriterionBox.tsx

```diff
@@ -0,0 +1,458 @@
+import Criterion, { newCriterion } from '../Criterion';
+
+const fadeIn = keyframes`
+  0% { opacity: 0 }
+  30% { opacity: 0.1 }
+  70% { opacity: 0.9 }
+  100% { opacity: 1 }
+`;
+
+const Root = styled(Box)`
+  display: flex;
+  height: 40px;
+  border-radius: 3px;
+  box-shadow: ${({ theme }) => theme.shadows[1]};
+  overflow: hidden;
+  margin-right: ${({ theme }) => theme.spacing(1)}px;
+  font-size: 1.1rem;
+  color: ${({ theme }) => theme.palette.common.white};
+  cursor: pointer;
+  & > *:hover {
+    opacity: 0.9;
+  }
+  animation: 0.25s 0s both ${fadeIn};
+`;
+
+const FocusedRoot = styled(Box)`
+  margin-right: ${({ theme }) => theme.spacing(2)}px;
+  position: relative;
+  animation: 0.25s 0s both ${fadeIn};
+  z-index: ${({ theme }) => theme.zIndex.modal};
+`;
+
+const DeleteButton = styled.button`
+  height: 100%;
+  width: 30;
+  color: ${({ theme }) => theme.palette.common.white};
+  background-color: ${({ theme }) => theme.palette.primary.main};
+  cursor: pointer;
+  border: none;
+  &:focus {
+    outline: none;
+  }
+`;
+
+const Input = styled.input.attrs(() => ({
+  'data-testid': 'criterion-input',
+}))`
+  width: 350px;
+  height: 40px;
+  padding: 10px;
+  box-sizing: border-box;
+  font-size: 1.1rem;
+`;
+
+interface CriterionBoxProps {
+  criteria: Criterion[];
+  criterion: Criterion;
+  serviceNames: string[];
+  remoteServiceNames: string[];
+  spanNames: string[];
+  autocompleteKeys: string[];
+  autocompleteValues: string[];
+  isLoadingServiceNames: boolean;
+  isLoadingRemoteServiceNames: boolean;
+  isLoadingSpanNames: boolean;
+  isLoadingAutocompleteValues: boolean;
+  isFocused: boolean;
+  onFocus: () => void;
+  onBlur: () => void;
+  onDecide: () => void;
+  onChange: (criterion: Criterion) => void;
+  onDelete: () => void;
+  loadAutocompleteValues: (autocompleteKey: string) => void;
+}
+
+const initialText = (criterion: Criterion) => {
+  if (criterion.key) {
+    if (criterion.value) {
+      return `${criterion.key}=${criterion.value}`;
+    }
+    return `${criterion.key}=`;
+  }
+  return '';
+};
+
+const CriterionBox: React.FC<CriterionBoxProps> = ({
+  criteria,
+  criterion,
+  serviceNames,
+  remoteServiceNames,
+  spanNames,
+  autocompleteKeys,
+  autocompleteValues,
+  isLoadingServiceNames,
+  isLoadingRemoteServiceNames,
+  isLoadingSpanNames,
+  isLoadingAutocompleteValues,
+  isFocused,
+  onFocus,
+  onBlur,
+  onDecide,
+  onChange,
+  onDelete,
+  loadAutocompleteValues,
+}) => {
+  const inputEl = useRef<HTMLInputElement>(null);
+
+  const [text, setText] = useState(initialText(criterion));
+  const [fixedText, setFixedText] = useState(initialText(criterion));
+
+  useMount(() => {
+    if (inputEl.current) {
+      inputEl.current.focus();
+    }
+  });
+
+  const prevIsFocused = useRef(isFocused);
+  useEffect(() => {
+    if (prevIsFocused.current && !isFocused) {
+      if (!fixedText) {
+        onDelete();
+        return;
+      }
+      let ss = fixedText.split('=');
```

Comment:
Can we use a more descriptive name than `ss`? Not sure what `ss` stands for here